mne_bids/copyfiles.py

```diff
@@ -339,6 +341,96 @@ def copyfile_brainvision(vhdr_src, vhdr_dest, anonymize=None, verbose=False):
+
+        `daysback` : int
+            Number of days by which to move back the recording date in time.
+            In studies with multiple subjects the relative recording date
+            differences between subjects can be kept by using the same number
+            of `daysback` for all subject anonymizations. `daysback` should be
+            great enough to shift the date prior to 1925 to conform with BIDS
+            anonymization rules.
+
+        `keep_his` : bool
+            By default (False), all subject information next to the recording
+            date will be overwritten as well. If True, keep subject information
+            apart from the recording date.
+
+    See Also
+    --------
+    mne.io.anonymize_info
+
+    """
+    # Ensure source & destination extensions are the same
+    fname_src, ext_src = _parse_ext(src)
+    fname_dest, ext_dest = _parse_ext(dest)
+    if ext_src != ext_dest:
+        raise ValueError(f'Need to move data with same extension'
+                         f' but got "{ext_src}", "{ext_dest}"')
+
+    # Copy data prior to any anonymization
+    sh.copyfile(src, dest)
+
+    # Anonymize EDF/BDF data, if requested
+    if anonymize is not None:
+        if ext_src == '.bdf':
+            raw = read_raw_bdf(dest, preload=False, verbose=0)
+        else:
+            raw = read_raw_edf(dest, preload=False, verbose=0)
+
+        # Get patient info, recording info, and recording date
+        with open(dest, 'rb') as f:
+            f.seek(8) # id_info field starts 8 bytes in
+            id_info = f.read(80).decode('ascii').rstrip()
+            rec_info = f.read(80).decode('ascii').rstrip()
+
+        # Parse metadata from file
+        if len(id_info) == 0 or len(id_info.split(' ')) != 4:
+            id_info = "X X X X"
+        if len(rec_info) == 0 or len(rec_info.split(' ')) != 5:
+            rec_info = "Startdate X X X X"
+        pid, sex, birthdate, name = id_info.split(' ')
+        startdate, admin_code, tech, equip = rec_info.split(' ')[1:5]
+
+        # Anonymize recording date, using 4-digit year from recording info
+        # if possible (regular EDF year only stores last two digits)
+        try:
+            actual_year = datetime.datetime.strptime(startdate, "%d-%b-%Y").year
+            newdate = raw.info['meas_date'].replace(year = actual_year)
+            raw.info['meas_date'] = newdate
+        except (ValueError, AttributeError):
+            pass
+        daysback, keep_his = _check_anonymize(anonymize, raw, '.edf')
+        info = anonymize_info(raw.info, daysback=daysback, keep_his=keep_his)
+        startdate = datetime.strftime(info['meas_date'], "%d-%b-%Y").upper()
+        meas_date = datetime.strftime(info['meas_date'], "%d.%m.%y")
+
+        # Anonymize ID info and write to file
+        if keep_his:
+            id_info = [pid, sex, "X", "X"]
+            rec_info = ["Startdate", startdate, admin_code, tech, equip]
```

Comment:
adding a comment here for future devs

```suggestion
            # always remove participant name and birthdate to be safe
            id_info = [pid, sex, "X", "X"]
            rec_info = ["Startdate", startdate, admin_code, tech, equip]
```

Reply:
actually, this :point_up: should also be mentioned in the docstr, that if anonymization happens, participant name and birthdate are always removed, no matter what.

Reply:
Good catch, added in the latest commit!